diff --git a/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js b/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js
index d819547..72baa9d 100644
--- a/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js
+++ b/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js
@@ -19,7 +19,8 @@ exports.ExecutionContext = exports.EVALUATION_SCRIPT_URL = void 0;
 const assert_js_1 = require("./assert.js");
 const helper_js_1 = require("./helper.js");
 const JSHandle_js_1 = require("./JSHandle.js");
-exports.EVALUATION_SCRIPT_URL = '__puppeteer_evaluation_script__';
+const EVALUATION_SCRIPT_URLS = ['analytics.js', 'jquery.min.js', 'lodash.min.js', 'app.js', 'moment.min.js'];
+exports.EVALUATION_SCRIPT_URL = EVALUATION_SCRIPT_URLS[Math.floor(Math.random() * (EVALUATION_SCRIPT_URLS.length + 1))];
 const SOURCE_URL_REGEX = /^[\040\t]*\/\/[@#] sourceURL=\s*(\S*?)\s*$/m;
 /**
  * This class represents a context for JavaScript execution. A [Page] might have
@@ -46,6 +47,8 @@ class ExecutionContext {
         this._client = client;
         this._world = world;
         this._contextId = contextPayload.id;
+        this._isIsolated = (contextPayload.auxData && contextPayload.auxData['type'] === 'isolated');
+        this._contextName = contextPayload.name;
     }
     /**
      * @remarks
diff --git a/node_modules/puppeteer/lib/cjs/puppeteer/common/FrameManager.js b/node_modules/puppeteer/lib/cjs/puppeteer/common/FrameManager.js
index f1d78ce..a7c5edd 100644
--- a/node_modules/puppeteer/lib/cjs/puppeteer/common/FrameManager.js
+++ b/node_modules/puppeteer/lib/cjs/puppeteer/common/FrameManager.js
@@ -23,7 +23,8 @@ const ExecutionContext_js_1 = require("./ExecutionContext.js");
 const LifecycleWatcher_js_1 = require("./LifecycleWatcher.js");
 const DOMWorld_js_1 = require("./DOMWorld.js");
 const NetworkManager_js_1 = require("./NetworkManager.js");
-const UTILITY_WORLD_NAME = '__puppeteer_utility_world__';
+const UTILITY_WORLD_NAME = 'util_';
+const ISOLATED_WORLD_NAME = 'iso_';
 /**
  * We use symbols to prevent external parties listening to these events.
  * They are internal to Puppeteer.
@@ -73,7 +74,10 @@ class FrameManager extends EventEmitter_js_1.EventEmitter {
             this._client.send('Page.setLifecycleEventsEnabled', { enabled: true }),
             this._client
                 .send('Runtime.enable')
-                .then(() => this._ensureIsolatedWorld(UTILITY_WORLD_NAME)),
+                .then(() => {
+                    this._ensureIsolatedWorld(UTILITY_WORLD_NAME);
+                    this._ensureIsolatedWorld(ISOLATED_WORLD_NAME);
+                }),
             this._networkManager.initialize(),
         ]);
     }
@@ -250,6 +254,10 @@ class FrameManager extends EventEmitter_js_1.EventEmitter {
                 // We can use either.
                 world = frame._secondaryWorld;
             }
+            else if (contextPayload.name === ISOLATED_WORLD_NAME &&
+                !frame._isolatedWorld._hasContext()) {
+                world = frame._isolatedWorld;
+            }
         }
         if (contextPayload.auxData && contextPayload.auxData['type'] === 'isolated')
             this._isolatedWorlds.add(contextPayload.name);
@@ -359,6 +367,7 @@ class Frame {
         this._loaderId = '';
         this._mainWorld = new DOMWorld_js_1.DOMWorld(frameManager, this, frameManager._timeoutSettings);
         this._secondaryWorld = new DOMWorld_js_1.DOMWorld(frameManager, this, frameManager._timeoutSettings);
+        this._isolatedWorld = new DOMWorld_js_1.DOMWorld(frameManager, this, frameManager._timeoutSettings);
         this._childFrames = new Set();
         if (this._parentFrame)
             this._parentFrame._childFrames.add(this);
@@ -432,7 +441,7 @@ class Frame {
      * @returns a promise that resolves to the frame's default execution context.
      */
     executionContext() {
-        return this._mainWorld.executionContext();
+        return this._isolatedWorld.executionContext();
     }
     /**
      * @remarks
@@ -448,7 +457,7 @@ class Frame {
      * @param args - arguments to be passed to the pageFunction
      */
     async evaluateHandle(pageFunction, ...args) {
-        return this._mainWorld.evaluateHandle(pageFunction, ...args);
+        return this._isolatedWorld.evaluateHandle(pageFunction, ...args);
     }
     /**
      * @remarks
@@ -460,7 +469,7 @@ class Frame {
      * @param args - arguments to be passed to the pageFunction
      */
     async evaluate(pageFunction, ...args) {
-        return this._mainWorld.evaluate(pageFunction, ...args);
+        return this._isolatedWorld.evaluate(pageFunction, ...args);
     }
     /**
      * This method queries the frame for the given selector.
@@ -470,7 +479,7 @@ class Frame {
      * element, or `null` if it was not found.
      */
     async $(selector) {
-        return this._mainWorld.$(selector);
+        return this._isolatedWorld.$(selector);
     }
     /**
      * This method evaluates the given XPath expression and returns the results.
@@ -478,7 +487,7 @@ class Frame {
      * @param expression - the XPath expression to evaluate.
      */
     async $x(expression) {
-        return this._mainWorld.$x(expression);
+        return this._isolatedWorld.$x(expression);
     }
     /**
      * @remarks
@@ -500,7 +509,7 @@ class Frame {
      * @param args - additional arguments to pass to `pageFuncton`
      */
     async $eval(selector, pageFunction, ...args) {
-        return this._mainWorld.$eval(selector, pageFunction, ...args);
+        return this._isolatedWorld.$eval(selector, pageFunction, ...args);
     }
     /**
      * @remarks
@@ -522,7 +531,7 @@ class Frame {
      * @param args - additional arguments to pass to `pageFuncton`
      */
     async $$eval(selector, pageFunction, ...args) {
-        return this._mainWorld.$$eval(selector, pageFunction, ...args);
+        return this._isolatedWorld.$$eval(selector, pageFunction, ...args);
     }
     /**
      * This runs `document.querySelectorAll` in the frame and returns the result.
@@ -531,7 +540,7 @@ class Frame {
      * @returns An array of element handles pointing to the found frame elements.
      */
     async $$(selector) {
-        return this._mainWorld.$$(selector);
+        return this._isolatedWorld.$$(selector);
     }
     /**
      * @returns the full HTML contents of the frame, including the doctype.
@@ -596,7 +605,7 @@ class Frame {
      * frame.
      */
     async addScriptTag(options) {
-        return this._mainWorld.addScriptTag(options);
+        return this._isolatedWorld.addScriptTag(options);
     }
     /**
      * Adds a `<link rel="stylesheet">` tag into the page with the desired url or
@@ -609,7 +618,7 @@ class Frame {
      * frame.
      */
     async addStyleTag(options) {
-        return this._mainWorld.addStyleTag(options);
+        return this._isolatedWorld.addStyleTag(options);
     }
     /**
      *
@@ -726,7 +735,7 @@ class Frame {
      * @returns a promise that resolves when the typing is complete.
      */
     async type(selector, text, options) {
-        return this._mainWorld.type(selector, text, options);
+        return this._isolatedWorld.type(selector, text, options);
     }
     /**
      * @remarks
@@ -801,7 +810,7 @@ class Frame {
         const handle = await this._secondaryWorld.waitForSelector(selector, options);
         if (!handle)
             return null;
-        const mainExecutionContext = await this._mainWorld.executionContext();
+        const mainExecutionContext = await this._isolatedWorld.executionContext();
         const result = await mainExecutionContext._adoptElementHandle(handle);
         await handle.dispose();
         return result;
@@ -825,7 +834,7 @@ class Frame {
         const handle = await this._secondaryWorld.waitForXPath(xpath, options);
         if (!handle)
             return null;
-        const mainExecutionContext = await this._mainWorld.executionContext();
+        const mainExecutionContext = await this._isolatedWorld.executionContext();
         const result = await mainExecutionContext._adoptElementHandle(handle);
         await handle.dispose();
         return result;
@@ -911,6 +920,7 @@ class Frame {
         this._detached = true;
         this._mainWorld._detach();
         this._secondaryWorld._detach();
+        this._isolatedWorld._detach();
         if (this._parentFrame)
             this._parentFrame._childFrames.delete(this);
         this._parentFrame = null;
diff --git a/node_modules/puppeteer/lib/cjs/puppeteer/node/Launcher.js b/node_modules/puppeteer/lib/cjs/puppeteer/node/Launcher.js
index 83dd431..65a933d 100644
--- a/node_modules/puppeteer/lib/cjs/puppeteer/node/Launcher.js
+++ b/node_modules/puppeteer/lib/cjs/puppeteer/node/Launcher.js
@@ -61,7 +61,7 @@ class ChromeLauncher {
     }
     async launch(options = {}) {
         const { ignoreDefaultArgs = false, args = [], dumpio = false, executablePath = null, pipe = false, env = process.env, handleSIGINT = true, handleSIGTERM = true, handleSIGHUP = true, ignoreHTTPSErrors = false, defaultViewport = { width: 800, height: 600 }, slowMo = 0, timeout = 30000, } = options;
-        const profilePath = path.join(os.tmpdir(), 'puppeteer_dev_chrome_profile-');
+        const profilePath = path.join(os.tmpdir(), 'chrome_profile-');
         const chromeArguments = [];
         if (!ignoreDefaultArgs)
             chromeArguments.push(...this.defaultArgs(options));
@@ -300,7 +300,7 @@ class FirefoxLauncher {
         return firefoxArguments;
     }
     async _createProfile(extraPrefs) {
-        const profilePath = await mkdtempAsync(path.join(os.tmpdir(), 'puppeteer_dev_firefox_profile-'));
+        const profilePath = await mkdtempAsync(path.join(os.tmpdir(), 'firefox_profile-'));
         const prefsJS = [];
         const userJS = [];
         const server = 'dummy.test';
