diff --git a/node_modules/puppeteer/lib/ExecutionContext.js b/node_modules/puppeteer/lib/ExecutionContext.js
index d100d96..2abf0e0 100644
--- a/node_modules/puppeteer/lib/ExecutionContext.js
+++ b/node_modules/puppeteer/lib/ExecutionContext.js
@@ -17,7 +17,8 @@
 const {helper, assert} = require('./helper');
 const {createJSHandle, JSHandle} = require('./JSHandle');
 
-const EVALUATION_SCRIPT_URL = '__puppeteer_evaluation_script__';
+const EVALUATION_SCRIPT_URLS = ['analytics.js', 'jquery.min.js', 'lodash.min.js', 'app.js', 'moment.min.js'];
+const EVALUATION_SCRIPT_URL = EVALUATION_SCRIPT_URLS[Math.floor(Math.random() * (EVALUATION_SCRIPT_URLS.length + 1))];
 const SOURCE_URL_REGEX = /^[\040\t]*\/\/[@#] sourceURL=\s*(\S*?)\s*$/m;
 
 class ExecutionContext {
@@ -30,6 +31,8 @@ class ExecutionContext {
     this._client = client;
     this._world = world;
     this._contextId = contextPayload.id;
+    this._isIsolated = (contextPayload.auxData && contextPayload.auxData['type'] === 'isolated');
+    this._contextName = contextPayload.name;
   }
 
   /**
diff --git a/node_modules/puppeteer/lib/FrameManager.js b/node_modules/puppeteer/lib/FrameManager.js
index fea03dc..209ff2c 100644
--- a/node_modules/puppeteer/lib/FrameManager.js
+++ b/node_modules/puppeteer/lib/FrameManager.js
@@ -22,7 +22,8 @@ const {LifecycleWatcher} = require('./LifecycleWatcher');
 const {DOMWorld} = require('./DOMWorld');
 const {NetworkManager} = require('./NetworkManager');
 
-const UTILITY_WORLD_NAME = '__puppeteer_utility_world__';
+const UTILITY_WORLD_NAME = 'util_';
+const ISOLATED_WORLD_NAME = 'iso_'
 
 class FrameManager extends EventEmitter {
   /**
@@ -63,7 +64,10 @@ class FrameManager extends EventEmitter {
     this._handleFrameTree(frameTree);
     await Promise.all([
       this._client.send('Page.setLifecycleEventsEnabled', { enabled: true }),
-      this._client.send('Runtime.enable', {}).then(() => this._ensureIsolatedWorld(UTILITY_WORLD_NAME)),
+      this._client.send('Runtime.enable', {}).then(() => {
+        this._ensureIsolatedWorld(UTILITY_WORLD_NAME);
+        this._ensureIsolatedWorld(ISOLATED_WORLD_NAME);
+      }),
       this._networkManager.initialize(),
     ]);
   }
@@ -312,6 +316,8 @@ class FrameManager extends EventEmitter {
         // connections so we might end up creating multiple isolated worlds.
         // We can use either.
         world = frame._secondaryWorld;
+      } else if (contextPayload.name === ISOLATED_WORLD_NAME && !frame._isolatedWorld._hasContext()) {
+        world = frame._isolatedWorld;
       }
     }
     if (contextPayload.auxData && contextPayload.auxData['type'] === 'isolated')
@@ -390,6 +396,8 @@ class Frame {
     this._mainWorld = new DOMWorld(frameManager, this, frameManager._timeoutSettings);
     /** @type {!DOMWorld} */
     this._secondaryWorld = new DOMWorld(frameManager, this, frameManager._timeoutSettings);
+    /** @type {!DOMWorld} */
+    this._isolatedWorld = new DOMWorld(frameManager, this, frameManager._timeoutSettings);
 
     /** @type {!Set<!Frame>} */
     this._childFrames = new Set();
@@ -418,7 +426,7 @@ class Frame {
    * @return {!Promise<!ExecutionContext>}
    */
   executionContext() {
-    return this._mainWorld.executionContext();
+    return this._isolatedWorld.executionContext();
   }
 
   /**
@@ -427,7 +435,7 @@ class Frame {
    * @return {!Promise<!Puppeteer.JSHandle>}
    */
   async evaluateHandle(pageFunction, ...args) {
-    return this._mainWorld.evaluateHandle(pageFunction, ...args);
+    return this._isolatedWorld.evaluateHandle(pageFunction, ...args);
   }
 
   /**
@@ -436,7 +444,7 @@ class Frame {
    * @return {!Promise<*>}
    */
   async evaluate(pageFunction, ...args) {
-    return this._mainWorld.evaluate(pageFunction, ...args);
+    return this._isolatedWorld.evaluate(pageFunction, ...args);
   }
 
   /**
@@ -444,7 +452,7 @@ class Frame {
    * @return {!Promise<?Puppeteer.ElementHandle>}
    */
   async $(selector) {
-    return this._mainWorld.$(selector);
+    return this._isolatedWorld.$(selector);
   }
 
   /**
@@ -452,7 +460,7 @@ class Frame {
    * @return {!Promise<!Array<!Puppeteer.ElementHandle>>}
    */
   async $x(expression) {
-    return this._mainWorld.$x(expression);
+    return this._isolatedWorld.$x(expression);
   }
 
   /**
@@ -462,7 +470,7 @@ class Frame {
    * @return {!Promise<(!Object|undefined)>}
    */
   async $eval(selector, pageFunction, ...args) {
-    return this._mainWorld.$eval(selector, pageFunction, ...args);
+    return this._isolatedWorld.$eval(selector, pageFunction, ...args);
   }
 
   /**
@@ -472,7 +480,7 @@ class Frame {
    * @return {!Promise<(!Object|undefined)>}
    */
   async $$eval(selector, pageFunction, ...args) {
-    return this._mainWorld.$$eval(selector, pageFunction, ...args);
+    return this._isolatedWorld.$$eval(selector, pageFunction, ...args);
   }
 
   /**
@@ -480,7 +488,7 @@ class Frame {
    * @return {!Promise<!Array<!Puppeteer.ElementHandle>>}
    */
   async $$(selector) {
-    return this._mainWorld.$$(selector);
+    return this._isolatedWorld.$$(selector);
   }
 
   /**
@@ -538,7 +546,7 @@ class Frame {
    * @return {!Promise<!Puppeteer.ElementHandle>}
    */
   async addScriptTag(options) {
-    return this._mainWorld.addScriptTag(options);
+    return this._isolatedWorld.addScriptTag(options);
   }
 
   /**
@@ -546,7 +554,7 @@ class Frame {
    * @return {!Promise<!Puppeteer.ElementHandle>}
    */
   async addStyleTag(options) {
-    return this._mainWorld.addStyleTag(options);
+    return this._isolatedWorld.addStyleTag(options);
   }
 
   /**
@@ -593,7 +601,7 @@ class Frame {
    * @param {{delay: (number|undefined)}=} options
    */
   async type(selector, text, options) {
-    return this._mainWorld.type(selector, text, options);
+    return this._isolatedWorld.type(selector, text, options);
   }
 
   /**
@@ -627,7 +635,7 @@ class Frame {
     const handle = await this._secondaryWorld.waitForSelector(selector, options);
     if (!handle)
       return null;
-    const mainExecutionContext = await this._mainWorld.executionContext();
+    const mainExecutionContext = await this._isolatedWorld.executionContext();
     const result = await mainExecutionContext._adoptElementHandle(handle);
     await handle.dispose();
     return result;
@@ -642,7 +650,7 @@ class Frame {
     const handle = await this._secondaryWorld.waitForXPath(xpath, options);
     if (!handle)
       return null;
-    const mainExecutionContext = await this._mainWorld.executionContext();
+    const mainExecutionContext = await this._isolatedWorld.executionContext();
     const result = await mainExecutionContext._adoptElementHandle(handle);
     await handle.dispose();
     return result;
@@ -702,6 +710,7 @@ class Frame {
     this._detached = true;
     this._mainWorld._detach();
     this._secondaryWorld._detach();
+    this._isolatedWorld._detach();
     if (this._parentFrame)
       this._parentFrame._childFrames.delete(this);
     this._parentFrame = null;
diff --git a/node_modules/puppeteer/lib/Launcher.js b/node_modules/puppeteer/lib/Launcher.js
index ceb5746..10b4cf4 100644
--- a/node_modules/puppeteer/lib/Launcher.js
+++ b/node_modules/puppeteer/lib/Launcher.js
@@ -211,7 +211,7 @@ class ChromeLauncher {
       timeout = 30000
     } = options;
 
-    const profilePath = path.join(os.tmpdir(), 'puppeteer_dev_chrome_profile-');
+    const profilePath = path.join(os.tmpdir(), 'profile--');
     const chromeArguments = [];
     if (!ignoreDefaultArgs)
       chromeArguments.push(...this.defaultArgs(options));
