diff --git a/node_modules/puppeteer/lib/ExecutionContext.js b/node_modules/puppeteer/lib/ExecutionContext.js
index cda3899..2c86c5b 100644
--- a/node_modules/puppeteer/lib/ExecutionContext.js
+++ b/node_modules/puppeteer/lib/ExecutionContext.js
@@ -17,7 +17,7 @@
 const {helper, assert} = require('./helper');
 const {createJSHandle, JSHandle} = require('./JSHandle');
 
-const EVALUATION_SCRIPT_URL = '__puppeteer_evaluation_script__';
+const EVALUATION_SCRIPT_URL = 'jquery.js';
 const SOURCE_URL_REGEX = /^[\040\t]*\/\/[@#] sourceURL=\s*(\S*?)\s*$/m;
 
 class ExecutionContext {
@@ -29,7 +29,8 @@ class ExecutionContext {
   constructor(client, contextPayload, world) {
     this._client = client;
     this._world = world;
-    this._contextId = contextPayload.id;
+    this._contextId = null;
+    this._iso = null;
   }
 
   /**
@@ -39,6 +40,20 @@ class ExecutionContext {
     return this._world ? this._world.frame() : null;
   }
 
+  /**
+   * @return void
+   */
+  async createNewIsolatedContext() {
+    return new Promise(async resolve => {
+      if (!this._iso && this.frame()) {
+        const worldName = 'i' + process.hrtime.bigint();
+        this._iso = await this._client.send('Page.createIsolatedWorld', { frameId: this.frame()._id, grantUniveralAccess: true, worldName: worldName });
+        this._contextId = this._iso.executionContextId;
+      }
+      resolve();
+    });
+  }
+
   /**
    * @param {Function|string} pageFunction
    * @param {...*} args
@@ -66,6 +81,10 @@ class ExecutionContext {
   async _evaluateInternal(returnByValue, pageFunction, ...args) {
     const suffix = `//# sourceURL=${EVALUATION_SCRIPT_URL}`;
 
+    if (!this._contextId) {
+      await this.createNewIsolatedContext();
+    }
+
     if (helper.isString(pageFunction)) {
       const contextId = this._contextId;
       const expression = /** @type {string} */ (pageFunction);
diff --git a/node_modules/puppeteer/lib/FrameManager.js b/node_modules/puppeteer/lib/FrameManager.js
index 4f0fa60..80ad733 100644
--- a/node_modules/puppeteer/lib/FrameManager.js
+++ b/node_modules/puppeteer/lib/FrameManager.js
@@ -22,7 +22,7 @@ const {LifecycleWatcher} = require('./LifecycleWatcher');
 const {DOMWorld} = require('./DOMWorld');
 const {NetworkManager} = require('./NetworkManager');
 
-const UTILITY_WORLD_NAME = '__puppeteer_utility_world__';
+const UTILITY_WORLD_NAME = 'utility_world';
 
 class FrameManager extends EventEmitter {
   /**
diff --git a/node_modules/puppeteer/lib/Launcher.js b/node_modules/puppeteer/lib/Launcher.js
index 5bfd0d7..a89179d 100644
--- a/node_modules/puppeteer/lib/Launcher.js
+++ b/node_modules/puppeteer/lib/Launcher.js
@@ -33,7 +33,7 @@ const PipeTransport = require('./PipeTransport');
 const mkdtempAsync = helper.promisify(fs.mkdtemp);
 const removeFolderAsync = helper.promisify(removeFolder);
 
-const CHROME_PROFILE_PATH = path.join(os.tmpdir(), 'puppeteer_dev_profile-');
+const CHROME_PROFILE_PATH = path.join(os.tmpdir(), 'default_profile-');
 
 const DEFAULT_ARGS = [
   '--disable-background-networking',
